timelock-impl/src/main/java/com/palantir/atlasdb/timelock/AsyncTimelockServiceImpl.java

```diff
@@ -257,24 +260,22 @@ public ListenableFuture<TimestampRange> getFreshTimestampsAsync(int timestampsTo
     }
 
     @Override
-    public ListenableFuture<TimestampLeaseResponse> acquireTimestampLease(
-            TimestampLeaseName timestampName, UUID requestId, int numFreshTimestamps) {
+    public ListenableFuture<TimestampLeaseResponses> acquireTimestampLease(
+            UUID requestId, Map<TimestampLeaseName, Integer> numFreshTimestamps) {
         long timestamp = timestampService.getFreshTimestamp();
 
         Leased<LockToken> leasedLock = lockService
-                .acquireTimestampLease(timestampName, requestId, timestamp)
+                .acquireTimestampLease(requestId, numFreshTimestamps.keySet(), timestamp)
                 .get();
-        long minLeased = lockService.getMinLeasedTimestamp(timestampName).orElse(timestamp);
 
```

Comment:
Let's add a comment here that it's crucial that the timestamps here are AFTER the fresh timestamp we're locking on.